conda_build/metadata.py

```diff
@@ -148,9 +148,12 @@ def parse(data):
     for field in FIELDS:
         if field not in res:
             continue
+        # ensure that empty fields are dicts (otherwise selectors can cause invalid fields)
+        if not res[field]:
+            res[field] = {}
         if not isinstance(res[field], dict):
-            raise RuntimeError("The %s field should be a dict, not %s" %
-                               (field, res[field].__class__.__name__))
+            raise RuntimeError("The %s field should be a dict, not %s in file %s." %
+                               (field, res[field].__class__.__name__, path))
```

Comment:
Here, I have extended the error message to also show the path of the meta.yaml. This is important in cases where conda build is applied to more than one recipe.